app/models/concerns/content_attribute_validator.rb

```diff
@@ -2,7 +2,7 @@ class ContentAttributeValidator < ActiveModel::Validator
   ALLOWED_SELECT_ITEM_KEYS = [:title, :value].freeze
   ALLOWED_CARD_ITEM_KEYS = [:title, :description, :media_url, :actions].freeze
   ALLOWED_CARD_ITEM_ACTION_KEYS = [:text, :type, :payload, :uri].freeze
-  ALLOWED_FORM_ITEM_KEYS = [:type, :placeholder, :label, :name, :options, :default].freeze
+  ALLOWED_FORM_ITEM_KEYS = [:type, :placeholder, :label, :name, :options, :default, :required, :regex].freeze
```

Comment:
let rename `:regex` to `:pattern`  since that confer with the standards.
Let's also introduce `:title` attribute so that the validation condition can be described by the API user

> Tip: Use the global title attribute to describe the pattern to help the user.
ref: https://www.w3schools.com/tags/att_pattern.asp